doc/source/reference/concepts.md

```diff
@@ -0,0 +1,98 @@
+# Seldon Core Concepts
+
+This page is a work in progress to provide concepts related to Seldon Core.
+
+This work is ongoing and we welcome feedback
+
+## Workflow
+#### Conceptual overview of workflows
+A workflow groups the components of a machine learning system into a logical pipeline. Sets of related components are grouped into graphs. The workflow contains the configuration of the components and the definitions of the inputs and outputs of the system, and of each component.
```

Comment:
Would a workflow also encompass the training side or just inference? If it's just inference, we could also add that we also refer to this one as an "inference graph" or "ml deployment".